Common/include/geometry/dual_grid/CEdge.hpp

```diff
@@ -0,0 +1,178 @@
+   * \return Number of nodes that set an edge (2).
+   */
+  inline unsigned short GetnNodes() const override { return 2; }
+
+  /*!
+   * \brief Compute Volume associated to each edge.
+   * \param[in] val_coord_Edge_CG - Coordinates of the centre of gravity of the edge.
+   * \param[in] val_coord_FaceElem_CG - Coordinates of the centre of gravity of the face of an element.
+   * \param[in] val_coord_Elem_CG - Coordinates of the centre of gravity of the element.
+   * \param[in] val_coord_Point - Coordinates of the point that form the control volume.
+   * \return Local volume associated to the edge.
+   */
+  su2double GetVolume(su2double *val_coord_Edge_CG, su2double *val_coord_FaceElem_CG, su2double *val_coord_Elem_CG, su2double *val_coord_Point) const;
+
+  /*!
+   * \overload
+   * \param[in] val_coord_Edge_CG - Coordinates of the centre of gravity of the edge.
+   * \param[in] val_coord_Elem_CG - Coordinates of the centre of gravity of the element.
+   * \param[in] val_coord_Point - Coordinates of the point that form the control volume.
+   * \return Local volume associated to the edge.
+   */
+  su2double GetVolume(su2double *val_coord_Edge_CG, su2double *val_coord_Elem_CG, su2double *val_coord_Point) const;
+
+  /*!
+   * \brief Set the face that correspond to an edge.
+   * \param[in] val_coord_Edge_CG - Coordinates of the centre of gravity of the edge.
+   * \param[in] val_coord_FaceElem_CG - Coordinates of the centre of gravity of the face of an element.
+   * \param[in] val_coord_Elem_CG - Coordinates of the centre of gravity of the element.
+   * \param[in] config - Definition of the particular problem.
+   * \return Compute the normal (dimensional) to the face that makes the control volume boundaries.
+   */
+  void SetNodes_Coord(su2double *val_coord_Edge_CG, su2double *val_coord_FaceElem_CG, su2double *val_coord_Elem_CG) override;
+
+  /*!
+   * \overload
+   * \brief Set the face that correspond to an edge.
+   * \param[in] val_coord_Edge_CG - Coordinates of the centre of gravity of the edge.
+   * \param[in] val_coord_Elem_CG - Coordinates of the centre of gravity of the element.
+   * \param[in] config - Definition of the particular problem.
+   * \return Compute the normal (dimensional) to the face that makes the contorl volume boundaries.
+   */
+  void SetNodes_Coord(su2double *val_coord_Edge_CG, su2double *val_coord_Elem_CG) override;
+
+  /*!
+   * \brief Copy the the normal vector of a face.
+   * \param[in] val_normal - Vector where the subroutine is goint to copy the normal (dimensional).
+   */
+  inline void GetNormal(su2double *val_normal) const override {
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      val_normal[iDim] = Normal[iDim];
+  }
+
+  /*!
+   * \brief Get the normal to a face of the control volume asociated with an edge.
+   * \return Dimensional normal vector, the modulus is the area of the face.
+   */
+  inline su2double *GetNormal(void) const override {  return Normal; }
```

Comment:
This is unusual, if the method is const it should not return a pointer that allows member data to be modified.